tools/rosbag/src/player.cpp

```diff
@@ -278,6 +284,20 @@ void Player::printTime()
     }
 }
 
+void Player::waitForSubscribers() const
+{
+    bool all_topics_subscribed = false;
+    std::cout << "Waiting for subscribers." << std::endl;
+    while (!all_topics_subscribed) {
+        all_topics_subscribed = true;
+        typedef std::pair<std::string, ros::Publisher> value_type;
+        foreach(const value_type& pub, publishers_) {
+            all_topics_subscribed &= pub.second.getNumSubscribers() > 0;
+        }
+    }
```

Comment:
Is this loop busy waiting? If yes, something less resource intensive might be good.